src/App.tsx

```diff
@@ -1,25 +1,46 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import './App.css';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+
+  const handlePageChange = (page: number) => {
+    setCurrentPage(page);
+  };
+
+  const handlePerPageChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(Number(event.target.value));
+    setCurrentPage(1);
+  };
+
+  const totalItems = items.length;
+  const startIndex = (currentPage - 1) * perPage;
+  const endIndex = startIndex + perPage;
```

Comment:
The calculation of 'startIndex' and 'endIndex' is correct for pagination. However, you should ensure that 'endIndex' does not exceed the total number of items to avoid empty pages.